NDTensors/ext/NDTensorsAMDGPUExt/adapt.jl

```diff
@@ -0,0 +1,19 @@
+using NDTensors.AMDGPUExtensions: AMDGPUExtensions, ROCArrayAdaptor
+using NDTensors.GPUArraysCoreExtensions: storagemode
+function AMDGPUExtensions.roc(xs)
+  return fmap(x -> adapt(ROCArrayAdaptor{AMDGPU.Runtime.Mem.HIPBuffer}(), x), xs)
+end
+
+function Adapt.adapt_storage(adaptor::ROCArrayAdaptor, xs::AbstractArray)
+  ElT = eltype(xs)
+  BufT = storagemode(adaptor)
+  N = ndims(xs)
+  return isbits(xs) ? xs : adapt(ROCArray{ElT,N,BufT}, xs)
```

Comment:
@kmp5VT this seems like a good place to use the new `TypeParameterAccessors` functionality.

Reply:
I updated this function and wrote an implementation with `TypeParameterAccessors` let me know if the way I wrote it is good or if you had something else in mind